src/executor/evaluate/mod.rs

```diff
@@ -395,11 +395,67 @@ async fn evaluate_function<'a, T: 'static + Debug>(
             Nullable::Null => Ok(Value::Null),
         }
         .map(Evaluated::from),
-        Function::Trim(expr) => match eval_to_str(expr).await? {
-            Nullable::Value(string) => Ok(Value::Str(string.trim().to_owned())),
-            Nullable::Null => Ok(Value::Null),
+        Function::Trim { expr, trim_where } => {
+            let expr_str = match eval_to_str(expr).await? {
+                Nullable::Value(str) => str,
+                Nullable::Null => return Ok(Value::Null).map(Evaluated::from),
+            };
+            let expr_str = expr_str.as_str();
+
+            if expr_str.is_empty() {
+                return Ok(Value::Str(expr_str.to_owned())).map(Evaluated::from);
+            }
+
+            match trim_where {
+                Some((trim_where_field, filter_chars)) => {
```

Comment:
Less indent would be better for readability.
How about this?

```rust
let ((trim_where_field, filter_chars)) = match trim_where {
    Some(trim_where) => trim_where,
    None => {
        return Ok(Evaluated::from(Value::Str(expr_str.trim().to_owned())));
    }
};

Reply:
The indentation is reduced and it looks very good!!!

Reply:
fixed it!